spike/CodeReaderClass.py

```diff
@@ -0,0 +1,151 @@
+class CodeParser():
+
+    #Parsing variables
+    PORT = ''
+    OPERATION = ''
+    NUMBER = ''
+
+    #Parsing libraries (or "keywords" if you wanna not sound like an arrogant prick)
+    OPERATIONS = ['+', '-', '*', '/', '=']
+    NUMBERS = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
+    WHITESPACE = ' '
+    PORTS = ["THROTTLE", "TURN"]
+
+    #Initialize the CodeParser
+    def __init__(self):
+        self.PATH = "Racer.txt"
+        self.SPEED = 0
+        self.ANGLE = 0
+        self.analyzer()
+
+    #Analyze
+    def analyzer(self):
+
+        #Open racer file
+        self.racer = open(self.PATH, "r")
+
+        #Break down lines into individual parts
+        self.lexeme = ''
+
+        #Check if file is readable and then read lines
+        if self.racer.mode == "r":
+            self.code = self.racer.readlines()
+
+            #For each line in the .txt file reset all stored variables then...
+            for self.line in self.code:
+                self.PORT = ''
+                self.OPERATION = ''
+                self.NUMBER = ''
+                self.lexeme = ''
+
+                #For each character in the line...
+                for self.i, self.char in enumerate(self.line):
+
+                    #If the character does not equal whitespace add it to the word
+                    if self.char != self.WHITESPACE:
+                       self.lexeme += self.char
```

Comment:
It looks like you are trying to grab the individual words out of the lines. In Python you can do this with `words = line.split()`. Then `words` is an array of the words in the line.